worker/index.js

```diff
@@ -70,9 +71,53 @@ async function work () {
     }
   }
 
+  async function subWrapper (subFn, ...eventFns) {
```

Comment:
I don't think we should be mixing wallet stuff into the index file. Separation of concerns and all that